modules/oe_content_event/src/Plugin/Validation/Constraint/EventFieldsRequiredValidator.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_content_event\Plugin\Validation\Constraint;
+
+use Drupal\node\NodeInterface;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Constraint;
+
+/**
+ * Checks if the event fields are provided if required.
+ */
+class EventFieldsRequiredValidator extends ConstraintValidator {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validate($node, Constraint $constraint) {
+    /* @var \Drupal\node\NodeInterface $node */
+    if (!isset($node) || $node->getType() !== 'oe_event') {
+      return;
+    }
+
+    // Validation of 'Organiser' fields.
+    if ($node->get('field_organiser_internal')->isEmpty() && $node->get('field_organiser_name')->isEmpty()) {
+      // Set message regarding empty organisation fields.
+      $violation = $this->context->buildViolation('You have to fill in at least one of fields @internal or @organiser_name', [
+        '@internal' => $node->getFieldDefinition('field_organiser_internal')->getLabel(),
+        '@organiser_name' => $node->getFieldDefinition('field_organiser_name')->getLabel(),
+      ]);
+      // Highlight empty 'Organiser name' field.
+      (clone $violation)
+        ->atPath('field_organiser_name')
+        ->addViolation();
+
+      // Highlight empty 'Internal organiser' field.
+      $violation
+        ->atPath('field_organiser_internal')
+        ->addViolation();
+    }
+
+    $online_required_fields = [
```

Comment:
Using the ticket OPENEUROPA-2188, we could remove this code and configure these fields normally.

Reply:
This is implemented according to documentation.

Reply:
As discussed with @nagyad this is missing a field in the check.